File: Lab 1/RideSharingApp__Strategy_Pattern_/src/CreditCardPayment.java
public class CreditCardPayment implements IPaymentMethod {
    private int CardNumber;
    private String CardHolderName;

    // Constructor
    public CreditCardPayment(int cardNumber, String cardHolderName) {
        this.CardNumber = cardNumber;
        this.CardHolderName = cardHolderName;
    }
    public int getCardNumber() {
        return CardNumber;
    }

    public String getCardHolderName() {
        return CardHolderName;
    }

    @Override
    public void ProcessPayment(double amount) {
        System.out.println("Processing credit card payment of " + amount);
        System.out.println("Card Number: " + CardNumber + ", Card Holder: " + CardHolderName);
    }
}

File: Lab_1_RideSharingApp/Lab_1_RideSharingApp/Payment/CreditCardPayment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_1_RideSharingApp
{
    public class CreditCardPayment : IPaymentMethod
    {
        public int CardNumber { get; private set; }
        public string CardHolderName { get; private set; }

        // Constructor
        public CreditCardPayment(int cardNumber, string cardHolderName)
        {
            CardNumber = cardNumber;
            CardHolderName = cardHolderName;
        }

        // Implementing IPaymentMethod interface
        public void ProcessPayment(double amount)
        {
            Console.WriteLine($"Processing credit card payment of {amount}.");
            Console.WriteLine($"Card Number: {CardNumber}, Card Holder: {CardHolderName}");
        }
    }
}
